lms/djangoapps/instructor/tests/test_registration_codes.py

```diff
@@ -0,0 +1,280 @@
+        self.assertEqual(response.status_code, 200, response.content)
+
+    def test_look_up_invalid_registration_code(self):
+        """
+        test lookup for the invalid registration code
+        """
+        data = {
+            'registration_code': 'invalid_reg_code'
+        }
+        response = self.client.post(self.lookup_code_url, data)
+        self.assertEqual(response.status_code, 400)
+        json_dict = json.loads(response.content)
+        message = _('The enrollment code ({code}) was not found for the {course_name} course.').format(
+            course_name=self.course.display_name, code=data['registration_code']
+        )
+        self.assertEqual(message, json_dict['message'])
+        self.assertFalse(json_dict['is_registration_code_valid'])
+        self.assertFalse(json_dict['is_registration_code_redeemed'])
+
+    def test_look_up_valid_registration_code(self):
+        """
+        test lookup for the valid registration code
+        and that registration code has been redeemed by user
+        and then mark the registration code as in_valid
+        when marking as invalidate, it also lookup for
+        registration redemption entry and also delete
+        that redemption entry and un_enroll the student
+        who used that registration code for their enrollment.
+        """
+        for i in range(2):
+            CourseRegistrationCode.objects.create(
+                code='reg_code{}'.format(i),
+                course_id=self.course.id.to_deprecated_string(),
+                created_by=self.instructor,
+                invoice=self.sale_invoice,
+                invoice_item=self.invoice_item,
+                mode_slug='honor'
+            )
+
+        reg_code = CourseRegistrationCode.objects.all()[0]
+        student = UserFactory()
+        enrollment = CourseEnrollment.enroll(student, self.course.id)
+
+        RegistrationCodeRedemption.objects.create(
+            registration_code=reg_code,
+            redeemed_by=student,
+            course_enrollment=enrollment
+        )
+
+        data = {
+            'registration_code': reg_code.code
+        }
+        response = self.client.post(self.lookup_code_url, data)
+        self.assertEqual(response.status_code, 200)
+        json_dict = json.loads(response.content)
+        self.assertTrue(json_dict['is_registration_code_valid'])
+        self.assertTrue(json_dict['is_registration_code_redeemed'])
+
+        # now mark that registration code as invalid
+        data = {
+            'registration_code': reg_code.code,
+            'action_type': 'invalidate_registration_code'
+        }
+        response = self.client.post(self.registration_code_detail_url, data)
+        self.assertEqual(response.status_code, 200)
+
+        json_dict = json.loads(response.content)
+        message = _('Action type {action_type} is a success.').format(action_type=data['action_type'])
```

Comment:
These four messages need to be updated so that they specifically describe the operation in human readable terms. I've provided suggestions based on what I think is going on in each situation; if I've guessed wrong, please use the suggestion that's applicable to the situation.

This enrollment code has been canceled. It can no longer be used.